server/src/services/metiers.service.ts

```diff
@@ -1,248 +1,190 @@
-        {
-          index: "domainesmetiers",
-          size: 20,
-          _source_includes: ["sous_domaine", "codes_romes", "codes_rncps"],
-          body: {
-            query: {
-              bool: {
-                must: terms,
-              },
-            },
-          },
-        },
-        DomainesMetiers
-      )
-
-      const labelsAndRomes: Omit<IMetierEnrichi, "romeTitles">[] = response.map((labelAndRome) => ({
-        label: labelAndRome._source.sous_domaine,
-        romes: labelAndRome._source.codes_romes,
-        rncps: labelAndRome._source.codes_rncps,
-        type: "job",
-      }))
+    const matchingMetier: IDomainesMetiers & { score?: number } = { ...metier }
+
+    searchableWeightedFields.map(({ field, score }) =>
+      regexes.map((regex) => {
+        const valueToTest = metier[field] instanceof Array ? metier[field].join(" ") : metier[field]
+        if (valueToTest.match(regex)) {
+          matchingMetier.score = score + (matchingMetier.score ?? 0)
+        }
+      })
+    )
 
-      return { labelsAndRomes }
-    } catch (error) {
-      const newError = Boom.internal("getting metiers from title romes and rncps")
-      newError.cause = error
-      throw newError
+    if (matchingMetier.score) {
+      results.push(matchingMetier)
     }
+  })
+
+  return results
+}
+
+const searchableWeightedDiplomaFields = [
+  { field: "intitule_long", score: 1 },
+  { field: "acronymes_intitule", score: 2 },
+]
+
+const filterDiplomas = async (regexes: any[]): Promise<(IDiplomesMetiers & { score?: number })[]> => {
+  if (cacheDiplomas.length === 0) {
+    await initializeCacheDiplomas()
   }
+
+  const results: (IDiplomesMetiers & { score?: number })[] = []
+
+  cacheDiplomas.map((diploma) => {
+    const matchingDiploma: IDiplomesMetiers & { score?: number } = { ...diploma }
+
+    searchableWeightedDiplomaFields.map(({ field, score }) =>
+      regexes.map((regex) => {
+        const valueToTest = diploma[field] instanceof Array ? diploma[field].join(" ") : diploma[field]
+        if (valueToTest.match(regex)) {
+          matchingDiploma.score = score + (matchingDiploma.score ?? 0)
+        }
+      })
+    )
+
+    if (matchingDiploma.score) {
+      results.push(matchingDiploma)
+    }
+  })
+
+  return results
 }
 
 /**
- * retourne une liste de métiers avec leurs codes romes et codes RNCPs associés. Le retour s'approche au mieux des critères.
- * @param {string} searchTerm : un préfixe, un mot ou un ensemble de préfixes ou mots sur lesquels fonder une recherche de métiers
- * @param {undefined | string} withRomeLabels : indique s'il faut que la fonction retourne également les labels associés aux romes
+ * construit les regex à partir du terme de recherche
  */
-const getLabelsAndRomes = async (searchTerm: string, withRomeLabels?: boolean): Promise<{ labelsAndRomes: IMetierEnrichi[] }> => {
-  try {
-    const terms: any[] = []
+const buildRegexes = (searchTerm: string) => {
+  const regexes: any[] = []
```

Comment:
```suggestion
  const regexes: RegExp[] = []
```
Est-ce que ça compile comme ça ?

Reply:
dans mon souvenir non c'est pour ça que je m'étais résigné à du any. mais je vais retenter.

Reply:
j'avais dû faire une typo. ca a marché ce coup ci.